lib/datasource/packagist/index.ts

```diff
@@ -199,10 +199,10 @@ async function packagistOrgLookup(name: string): Promise<ReleaseResult> {
   if (cachedResult) {
     return cachedResult;
   }
+
   let dep: ReleaseResult = null;
   const regUrl = 'https://packagist.org';
-  const pkgUrl = URL.resolve(regUrl, `/p/${name}.json`);
-  // TODO: fix types (#9610)
+  const pkgUrl = URL.resolve(regUrl, `/p2/${name}.json`);
```

Comment:
Please use `joinUrlParts` from `utils/url`.

This is all we need to change? No result object changes?

Then we don't need to change the cache key, as they seem to be compatible.

Reply:
I could only see 3 changes in the new API.
1. In the URL `p2` instead of `p`. In the docs, the URL used is
` https://repo.packagist.org/p2/monolog/monolog.json`, I didn't change the URL since both formats work fine.

2. The versions list is returned in an array instead of an object.
-> Due to the new return object containing an array, it seemed like we might have to change the logic for `extractDepReleases` but the way the fn is written handles the array as well.

3. It is in a compressed format for efficiency which requires you to use [link](https://github.com/composer/metadata-minifier/blob/main/src/MetadataMinifier.php#L17) from the package to restore it to the full data.
-> After comparing the returned object from old and new API, info like name, source, description..were being repeated in each array element in the older version, in the new API and this info is only found in the first element of the `packages` array.

4. The p2/$vendor/$package.json file contains only tagged releases. If you want to fetch information about branches (i.e. dev versions) you need to download p2/$vendor/$package~dev.json.
-> Want to discuss this one. In the old code, we are storing the dev versions as well. To Implement we fetch data and push it to the result objects `packages[nameOfDep]` field.